### DISCUSSION TITLE
Van Element 2.0

### DISCUSSION THREAD

**Discussion by Atmos4** [vanjs-org/van#345] [Show and tell]
I am happy to announce `vanjs-element@2.0.0` ☀️

The only breaking change is the removal of the `children` prop, which didn't work well and felt a bit awkward to use.

The `mount` function can now be called many times, which allows for the creation of reusable hooks 🥳

Check out the [main repository](https://github.com/Atmos4/van-element) for more information!

**Comment by Tao-VanJS** [comment]
Congratulations to the launch!

**Comment by sysmat** [comment]
how to use it inside van app?

**Reply by Atmos4** [reply]
@sysmat I am not quite sure what you are asking exactly :)

You can use Van Elements (and any custom elements) in VanJS with this syntax: `van.tags["my-custom-element"]`.

**Reply by sysmat** [reply]
- it renders nothing(no browser error) using https://github.com/Atmos4/van-element?tab=readme-ov-file#usage

```typescript
define('custom-counter', () => {
  const counter = van.state(0);
  return div(
    slot(),
    counter,
    button({ class: buttonCss, onclick: () => ++counter.val }, '+'),
    button({ class: buttonCss, onclick: () => --counter.val }, '-')
  );
});

div(
    {
      class: 'flex flex-wrap items-center justify-between max-w-screen-xl px-4 mx-auto',
    },    
    van.tags['my-custom-element']    
  );

```

**Reply by Atmos4** [reply]
Haha that's because you need to use a tag you defined 😅 so for example here if you define an element called `custom-counter`, you will be able to access it with `van.tags["custom-counter"]`

**Reply by sysmat** [reply]
@Atmos4 thx, but tailwind css not working

```typescript
export const buttonCss =
	'px-4 py-1 text-sm text-[#E65A00] font-semibold rounded-full border border-orange-200 hover:text-white hover:bg-orange-600 hover:border-transparent focus:outline-none focus:ring-2 focus:ring-purple-600 focus:ring-offset-2';
```

**Reply by Atmos4** [reply]
That's because by default, Van Element will enable the shadow DOM, which provides isolation from global styles. You can disable it like this:
```js
vane.define("custom-counter", () => {
  ...
}, false);
```
Keep in mind that if you disable it, you cannot use slots anymore as this mechanism is tied to the Shadow DOM.

**Reply by sysmat** [reply]
@Atmos4 thx
